wandb/sdk/interface/interface_sock.py

```diff
@@ -0,0 +1,58 @@
+"""InterfaceSock - Derived from InterfaceShared using a socket to send to internal thread
+
+See interface.py for how interface classes relate to each other.
+
+"""
+
+import logging
+from typing import Any, Optional
+from typing import TYPE_CHECKING
+
+
+from .interface_shared import InterfaceShared
+from .message_future import MessageFuture
+from .router_sock import MessageSockRouter
+from ..lib.sock_client import SockClient
+
+
+if TYPE_CHECKING:
+    from wandb.proto import wandb_internal_pb2 as pb
+    from ..wandb_run import Run
+
+
+logger = logging.getLogger("wandb")
+
+
+class InterfaceSock(InterfaceShared):
+    _stream_id: Optional[str]
+    _sock_client: SockClient
+
+    def __init__(self, sock_client: SockClient) -> None:
+        self._sock_client = sock_client
+        super(InterfaceSock, self).__init__()
```

Comment:
why this order?
```suggestion
        self._sock_client = sock_client
        super().__init__()
```

Reply:
The order is required... the base class initializes the router by calling the abstract method `_init_router` which needs the _sock_client to be initialized already.

I changed to the python3 style super(),  python2 habits are hard to shed.

Added comment

Reply:
There are other places where this order seems to appear. 

I found a few python libraries that convert py2 -> py3, we might want to run them at some point, since we are not supporting py2 anymore, what do you think?

Reply:
Not now.  that should be done a a separate pr.

Reply:
Oh yeah, didn't mean in this PR. meant in general as part of our clean up of the client library.